### ISSUE TITLE
The difference between Loss( f(x0+t_{n+1}*z) , f(x0+t_{n}*z) ) and Loss( f(x0+t_{n+1}*z), x0 )? 

### ISSUE THREAD

**Issue by ZGCTroy** [openai/consistency_models#11]
1. In CT, would it be acceptable to use Loss( f(x0+t_{n+1}*z), x0 ) in place of Loss( f(x0+t_{n+1}*z) , f(x0+t_{n}*z) ) ? 
2. I would like to know if doing so would still comply with the principles of consistency models, or would it completely deviate or only result in a slight decrease in performance?

**Comment by thorinf** [comment]
I think this might just make it a score-based model. Like Yang Songs earlier work, or instead very like [EDM](https://arxiv.org/pdf/2206.00364.pdf). The `+1` wouldn't really matter I feel.

**Comment by 1999kevin** [comment]
Or is it possible to use both Loss( f(x0+t_{n+1}*z), x0 ) and Loss( f(x0+t_{n+1}*z) , f(x0+t_{n}*z) )? The first term enforces the similarity with the ground truth images, while the second term enforces the consistency. Simply using the consistency term would in fact decrease the generation performance or lead to longer training.

**Comment by thorinf** [comment]
Sorry, I forgot to get back to this one. 

Definitely possibly, might train better images. I might try it.

**Comment by thorinf** [comment]
Possibly could yield better training. It does seem that EDM can outperform CD and CT consistency models on FID for CIFAR and LSUN. I think a good question to ask is why is self-consistency useful? Is it worth the additional forward pass to use it? Time to train EDM will be lower.

**Comment by jsttlgdkycy** [comment]
loss(x0+t_{n+1}*z), x0) can only enforce the model to learn the expectation of x0, not the target point of the current trajectory. It is  a wrong loss term for one step diffusion

**Comment by thorinf** [comment]
> loss(x0+t_{n+1}*z), x0) can only enforce the model to learn the expectation of x0, not the target point of the current trajectory. It is  a wrong loss term for one step diffusion

Yes, although you can target the GT I think this would actually make the task harder. By trying to target the trajectory you are smoothing the data manifold which makes training easier.
